### PR TITLE
Add tests for ResultTableStore and Shared/Index.ts

### PR DESCRIPTION
**Pull request by mukul-ms**
Added tests to meet our initial goal of >= 80% test coverage for tableStore, resultTableStore and shared/index.ts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -155,5 +155,8 @@
     "optionalDependencies": {
         "@microsoft/sarif-multitool-darwin": "2.2.5",
         "@microsoft/sarif-multitool-win32": "2.2.5"
+    },
```
Comment (mukul-ms) [inline package.json:158]: Already put this in the nyc config, will remove it here

File: src/panel/resultTableStore.spec.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        assert.strictEqual(resultTableStore.columns.length, 6);
+        assert.strictEqual(resultTableStore.columns[0].name, 'Line')
+        assert.strictEqual(resultTableStore.columns[1].name, 'File')
+        assert.strictEqual(resultTableStore.columns[2].name, 'Message')
+        assert.strictEqual(resultTableStore.columns[3].name, 'Baseline')
+        assert.strictEqual(resultTableStore.columns[4].name, 'Suppression')
+        assert.strictEqual(resultTableStore.columns[5].name, 'Rule')
+    });
+});
+r
```
Comment (mukul-ms) [inline src/panel/resultTableStore.spec.ts:35]: weird, would fix this too

**Comment by mukul-ms** [comment]
> Just the `nyc` settings?

Fixed

File: package.json (hunk 1 above)
Comment (mukul-ms) [reply package.json]: Fixed

File: src/panel/resultTableStore.spec.ts (hunk 2 above)
Comment (mukul-ms) [reply src/panel/resultTableStore.spec.ts]: fixed

File: src/panel/resultTableStore.spec.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    const filtersSource = {
+        keywords: '',
+        filtersRow: filtersRow,
+        filtersColumn: filtersColumn
+    };
+
+    it('should verify the default columns', () => {
+        const resultTableStore = new ResultTableStore('File', result => result._relativeUri, resultsSource, filtersSource, selection);
+        assert.strictEqual(resultTableStore.columns.length, 6);
+        resultTableStore.columns.map((col) => assert.strictEqual(['Line', 'File', 'Message', 'Baseline', 'Suppression', 'Rule'].includes(col.name), true));
```
Comment (jeffersonking) [inline src/panel/resultTableStore.spec.ts:24]: I think there is still room for simplification here.

File: src/panel/resultTableStore.spec.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+describe('ResultTableStore', () => {
+    const resultsSource = { results: log.runs![0].results! };
+    const selection = observable.box<Row | undefined>(undefined);
+    const filtersSource = {
+        keywords: '',
+        filtersRow: filtersRow,
+        filtersColumn: filtersColumn
+    };
+
+    it('should verify the default columns', () => {
```
Comment (jeffersonking) [inline src/panel/resultTableStore.spec.ts:21]: Your `describe`+`it` is effectively saying: "ResultTableStore should verify the default columns".

File: src/test/mockResultTableStore.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                    artifactLocation: {
+                        uri: '/folder/file_3.txt',
+                    }
+                }
+            }]
+        }]
+    }]
+} as Log;
+
+export const filtersRow = {
```
Comment (jeffersonking) [inline src/test/mockResultTableStore.ts:85]: I think you can use the definition from `shared/index.ts`.

File: src/panel/resultTableStore.spec.ts (hunk 3 above)
Comment (mukul-ms) [reply src/panel/resultTableStore.spec.ts]: Fixed

File: src/test/mockResultTableStore.ts (hunk 5 above)
Comment (mukul-ms) [reply src/test/mockResultTableStore.ts:85]: fixed

File: src/panel/resultTableStore.spec.ts (hunk 4 above)
Comment (mukul-ms) [reply src/panel/resultTableStore.spec.ts]: attempted to fix

**Comment by jeffersonking** [comment]
Looks like my approval was "stale". Looks like the majority of the files were added after that review. Still had a few comments so I'll add them here in this PR despite it already being merged.

File: src/test/mockResultTableStore.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (jeffersonking) [inline src/test/mockResultTableStore.ts:1]: Any reason for organizing/locating this file as `src/test/mockResultTableStore.ts`?

File: src/test/mockResultTableStore.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+// Licensed under the MIT License.
+
+import { Log } from 'sarif';
+
+export const log = {
```
Comment (jeffersonking) [inline src/test/mockResultTableStore.ts:6]: Reason for having the whole log? Only the `results` seem to be used.

File: src/test/mockResultTableStore.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+                    }
+                }
+            }]
+        }, {
+            message: {
+                text: 'Message 6'
+            },
+            level: 'none',
+            baselineState: 'new',
+            _suppression: 'not suppressed',
```
Comment (jeffersonking) [inline src/test/mockResultTableStore.ts:73]: Significance of `level`, `baselineState`, and `_suppression`?

File: src/test/mockResultTableStore.ts [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+import { Log } from 'sarif';
+
+export const log = {
+    version: '2.1.0',
+    runs: [{
+        tool: {
+            driver: { name: 'Driver' }
+        },
+        results: [{
```
Comment (jeffersonking) [inline src/test/mockResultTableStore.ts:12]: Purpose for having 6 results? If 6 results are really necessary, any way to reduce the repetition?

File: src/shared/index.spec.ts [hunk 10]
```diff
@@ -53,6 +53,16 @@ describe('augmentLog', () => {
     });
 });
 
+describe('decodeFileUri', () => {
+    it(`decodes the 'file' uri schemes`, () => {
```
Comment (jeffersonking) [inline src/shared/index.spec.ts:57]: Backticks should be reserved for template literals. If you want to use a quote in the string, you can use double-quotes.

**Comment by mukul-ms** [comment]
> Looks like my approval was "stale". Looks like the majority of the files were added after that review. Still had a few comments so I'll add them here in this PR despite it already being merged.

Oh weird. Sorry I saw 2 approvals and assumed the 2nd one was after our 1:1 chat. I should look into how to reset approvals if newer changes are checked in

File: src/test/mockResultTableStore.ts (hunk 7 above)
Comment (mukul-ms) [reply src/test/mockResultTableStore.ts:6]: pushing a fix in this PR: https://github.com/microsoft/sarif-vscode-extension/pull/304 👍

File: src/test/mockResultTableStore.ts (hunk 9 above)
Comment (mukul-ms) [reply src/test/mockResultTableStore.ts:12]: pushing a fix in this PR: https://github.com/microsoft/sarif-vscode-extension/pull/304 👍

File: src/shared/index.spec.ts (hunk 10 above)
Comment (mukul-ms) [reply src/shared/index.spec.ts:57]: pushing a fix in this PR: https://github.com/microsoft/sarif-vscode-extension/pull/304 👍

**Pull request merged**